test/suite/test_hs11.py

```diff
@@ -75,9 +75,9 @@ def test_non_ts_updates_clears_hs(self):
         cursor = self.session.open_cursor(uri)
         for ts in range(1, 5):
             for i in range(1, self.nrows):
-                self.session.begin_transaction()
-                cursor[self.create_key(i)] = value1
-                self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(ts))
+                for retry in self.retry():
+                    with retry.transaction(commit_timestamp = ts):
+                        cursor[self.create_key(i)] = value1
```

Comment:
I cannot say that this format is super-intuitive to me. I would not be able to write fresh, new test code using transactions without copy/paste from something existing.